passes/source2il.nim

```diff
@@ -405,6 +405,37 @@ proc exprToIL(c; t: InTree, n: NodeIndex, bu, stmts): SemType =
       # it's a unit value
       bu.add Node(kind: IntVal)
       result = prim(tkUnit)
+  of SourceKind.TupleAccess:
```

Comment:
Would `FieldAccess` work better? It can apply to both tuples and objects.

---

#### Brain dump about compound types and fields

This is based off of NimSkull, but here are the types and how we access them and various relevant properties.

|Type|Operator|Action|Subcomponent Name|Ordered|Width|
|-----|---------|------|---------------|--------|-----|
|array|`[]`|index|element|yes|fixed|
|tuple|`[]`|index|element|yes|variable|
|tuple|`.`|access|field|yes|variable|
|object|`.`|access|field|no|variable|

I've also seen the term "subscript" used for array access, but I consider that to be somewhat off; although the advantage is that it frees up the term `index` to act as a generalization over `subscript` and `field access`. As long as we're not looking to unify access across array, object, and tuples, then I think `FieldAccess` works, especially if we're considering mapping it to the `.` operator in some top level syntax.